Modules/Core/Transform/test/itkResampleInPlaceImageFilterTest.cxx

```diff
@@ -0,0 +1,145 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkResampleInPlaceImageFilter.h"
+
+#include "itkImage.h"
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkVersorRigid3DTransform.h"
+#include "itkImageRegionConstIterator.h"
+#include "itkTestingMacros.h"
+
+#include <iostream>
+
+// return true if it fails the validation
+inline bool
+Validate(double input, double desired, double tolerance)
+{
+  return std::abs<double>(input - desired) > tolerance * std::abs<double>(desired);
+}
+
+// Returns true if images are different
+template <typename ImageType>
+bool
+imagesDifferent(ImageType * baselineImage, ImageType * outputImage)
```

Comment:
Isn't the testing image comparison method enough in this case? If not, should this be added to the testing framework?

Reply:
This test can probably be further simplified. I was trying to reuse it, not spend time rewriting it completely. You are welcome to do so in a follow-up PR.